### PR TITLE
Feat/import vtp

### PR DESCRIPTION
**Pull request by panquez**

### CODE REVIEW

File: src/geode/io/mesh/vtp_input.cpp [hunk 1]
```diff
@@ -0,0 +1,424 @@
...
+        }
+
+    private:
+        void read_root_attributes()
+        {
+            OPENGEODE_EXCEPTION(
+                strcmp( root_.attribute( "type" ).value(), "PolyData" ) == 0,
+                "[VTPInput::read_root_attributes] VTK File type should be "
+                "PolyData" );
+            if( strcmp( root_.attribute( "byte_order" ).value(), "BigEndian" )
```
Comment (BotellaA) [inline src/geode/io/mesh/vtp_input.cpp:98]: Is this better https://github.com/abseil/abseil-cpp/blob/master/absl/strings/match.h

File: src/geode/io/mesh/vtp_input.cpp [hunk 2]
```diff
@@ -0,0 +1,424 @@
...
+                compressed_ = true;
+            }
+        }
+
+        void read_polydata( const pugi::xml_node& polydata )
+        {
+            for( const auto& piece : polydata.children( "Piece" ) )
+            {
+                const auto nb_polygons =
+                    std::stoul( piece.attribute( "NumberOfPolys" ).value() );
```
Comment (BotellaA) [inline src/geode/io/mesh/vtp_input.cpp:122]: same question here https://github.com/abseil/abseil-cpp/blob/master/absl/strings/numbers.h

File: src/geode/io/mesh/vtp_input.cpp [hunk 3]
```diff
@@ -0,0 +1,424 @@
...
+                reinterpret_cast< const geode::index_t* >(
+                    bytes.c_str() ); // should be unsigned long if UInt64
+            const auto nb_data_blocks = fixed_header_values[0];
+            if( nb_data_blocks == 0 )
+            {
+                return std::vector< T >{};
+            }
+            const auto uncompressed_block_size = fixed_header_values[1];
+            const auto nb_characters =
+                std::ceil( nb_data_blocks * 32. / 24. ) * 4;
```
Comment (BotellaA) [inline src/geode/io/mesh/vtp_input.cpp:183]: add a comment to explain this

File: src/geode/io/mesh/vtp_input.cpp [hunk 4]
```diff
@@ -0,0 +1,424 @@
...
+            auto optional_header = clean_input.substr( 16, nb_characters );
```
Comment (BotellaA) [inline src/geode/io/mesh/vtp_input.cpp:184]: What is 16? maybe sizeof(XXX)?

File: src/geode/io/mesh/vtp_input.cpp [hunk 5]
```diff
@@ -0,0 +1,424 @@
...
+                reinterpret_cast< const unsigned char* >( bytes.c_str() );
+
+            std::vector< T > result;
+            result.reserve( std::ceil(
+                nb_data_blocks * uncompressed_block_size / sizeof( T ) ) );
+            geode::index_t cur_data_offset{ 0 };
+            for( const auto b : geode::Range{ nb_data_blocks } )
+            {
+                const auto compressed_data_length =
+                    3 * sum_compressed_block_size;
```
Comment (BotellaA) [inline src/geode/io/mesh/vtp_input.cpp:222]: Magic numbers everywhere :p

File: src/geode/io/mesh/vtp_input.cpp [hunk 6]
```diff
@@ -0,0 +1,424 @@
...
+                size_t decompressed_data_length =
+                    nb_data_blocks * uncompressed_block_size;
+                std::unique_ptr< uint8_t > decompressed_data_bytes{
```
Comment (BotellaA) [inline src/geode/io/mesh/vtp_input.cpp:225]: Use a fixedArray since you know the size and the method `data()` to get the pointer.

File: src/geode/io/mesh/vtp_input.cpp [hunk 7]
```diff
@@ -0,0 +1,424 @@
...
+#include <zlib-ng.h>
+
+#include <geode/geometry/point.h>
+
+#include <geode/mesh/builder/polygonal_surface_builder.h>
+#include <geode/mesh/core/geode_polygonal_surface.h>
+
+namespace
+{
+    absl::string_view remove_spaces( absl::string_view in )
```
[the remark this answers is not in the document]
Comment (BotellaA) [reply src/geode/io/mesh/vtp_input.cpp:42]: https://github.com/abseil/abseil-cpp/blob/master/absl/strings/ascii.h#L225

[the remark this answers is not in the document]
Comment (BotellaA) [reply src/geode/io/mesh/vtp_input.cpp:42]: or https://github.com/abseil/abseil-cpp/blob/master/absl/strings/ascii.h#L237

File: src/geode/io/mesh/vtp_input.cpp [hunk 8]
```diff
@@ -0,0 +1,424 @@
...
+                        connectivity_values );
+                }
+            }
+            return get_polygon_vertices( connectivity, offsets );
+        }
+
+        template < typename T_in, typename T_out >
+        std::vector< T_out > cast_data_array( const std::vector< T_in >& data )
+        {
+            std::vector< T_out > casted;
```
Comment (BotellaA) [inline src/geode/io/mesh/vtp_input.cpp:360]: FixedArray?

File: src/geode/io/mesh/vtp_input.cpp (hunk 4 above)
Comment (panquez) [reply src/geode/io/mesh/vtp_input.cpp:184]: Yes I add comments to explain all these magical numbers

File: src/geode/io/mesh/vtp_input.cpp [hunk 9]
```diff
@@ -159,8 +142,12 @@ namespace
         template < typename T >
         std::vector< T > decode( absl::string_view input )
         {
-            auto clean_input = remove_spaces( input );
-            auto fixed_header = clean_input.substr( 0, 16 );
+            auto clean_input = absl::StripAsciiWhitespace( input );
+            const geode::index_t fixed_header_length{
```
Comment (BotellaA) [inline src/geode/io/mesh/vtp_input.cpp:146]:
```suggestion
            constexpr geode::index_t fixed_header_length{
```

File: src/geode/io/mesh/vtp_input.cpp [hunk 10]
```diff
@@ -219,20 +210,20 @@ namespace
             for( const auto b : geode::Range{ nb_data_blocks } )
             {
                 const auto compressed_data_length =
-                    3 * sum_compressed_block_size;
+                    // 3 * sum_compressed_block_size;
```
Comment (BotellaA) [inline src/geode/io/mesh/vtp_input.cpp:213]: Why did you remove 3* ?

File: src/geode/io/mesh/vtp_input.cpp [hunk 11]
```diff
@@ -390,11 +380,12 @@ namespace
             std::istringstream iss( data.data() );
```
Comment (BotellaA) [inline src/geode/io/mesh/vtp_input.cpp:380]: Maybe you don't need an iss with split functions https://github.com/abseil/abseil-cpp/blob/master/absl/strings/str_split.h#L493

Comment (BotellaA) [reply src/geode/io/mesh/vtp_input.cpp:380]: https://github.com/abseil/abseil-cpp/blob/master/absl/strings/str_split.h#L475

File: src/geode/io/mesh/vtp_input.cpp (hunk 10 above)
Comment (panquez) [reply src/geode/io/mesh/vtp_input.cpp:213]: It was a mistake
